source/opt/inline_pass.cpp

```diff
@@ -284,333 +670,60 @@ bool InlinePass::GenInlineCode(
     if (rid != 0) callee_result_ids.insert(rid);
   });
```

Comment:
I think we can change this to immediately assign new ids to `callee_result_ids`.  Then we can delete the references to `callee_result_ids`, since it will just be redundant information.  This could also help us simplify the code.  The only reason inline fails is because we run out of ids.  If we limit the places were we take the next id, then we do not have to check for failures in as many places.

If for some reason we cannot assign new ids right away, we can map all of the result ids to 0 initially.  Then that turns into the clue that it is a forward reference.  Either way, we do not need `callee_result_ids`.